openpype/hosts/maya/plugins/publish/extract_xgen_cache.py

```diff
@@ -0,0 +1,61 @@
+import os
+
+from maya import cmds
+
+import avalon.maya
+import openpype.api
+
+
+class ExtractXgenCache(openpype.api.Extractor):
+    """Produce an alembic of just xgen interactive groom
+
+    """
+
+    label = "Extract Xgen ABC Cache"
+    hosts = ["maya"]
+    families = ["xgen"]
+    optional = true
+
+    def process(self, instance):
+
+        # Collect the out set nodes
+        out_descriptions = [(node for node in instance
+                            if cmds.nodeType(node) == "xgmSplineDescription")]
```

Comment:
This doesn't work later on lien 48, just drop the generator expression and use only list comprehension:
```suggestion
        out_descriptions = [node for node in instance
                            if cmds.nodeType(node) == "xgmSplineDescription"]
```